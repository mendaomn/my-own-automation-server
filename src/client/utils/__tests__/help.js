const fs = require('fs')
const path = require('path')

const helpText = fs.readFileSync(path.join(__dirname, '..', 'help.txt'), 'utf8')

test('should contain help text for the CLI', () => {
  expect(helpText).toMatchSnapshot()
});
