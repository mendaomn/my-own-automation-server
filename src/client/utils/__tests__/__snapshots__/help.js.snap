// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should contain help text for the CLI 1`] = `
"yaci-client is a CLI for accessing the yaci-server

Be sure to have a running yaci-server instance

Commands:
  run ----------------- runs specified pipeline 
    args:
      <pipeline-name>   name of the pipeline to run 
    opts:
      --watch, -w       runs check in watch mode


  upload -------------- uploads a pipeline
    args: 
      <pipeline-name>   name of the pipeline
      <file-path>       path to the Makefile
    opts:
      --run, -r         run the pipeline right after uploading it
      --watch, -w       used along with --run, runs check in watch mode


  list ---------------- lists all pipelines


  check --------------- displays status of specified run
    args:
      <run-id>
    opts:
      --watch, -w       watch mode
      

Usage: 
  yaci <command> <args>

Use \\"yaci --help\\" to print this help

Examples:
  yaci upload staging ./pipelines/staging
  yaci run staging -w"
`;
